ci/build_wheel_libraft.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="libraft"
+package_dir="python/libraft"
+
+rapids-logger "Generating build requirements"
+matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file-key "py_build_${package_name}" \
+  --file-key "py_rapids_build_${package_name}" \
+  --matrix "${matrix_selectors}" \
+| tee /tmp/requirements-build.txt
+
+rapids-logger "Installing build requirements"
+python -m pip install \
+    -v \
+    --prefer-binary \
+    -r /tmp/requirements-build.txt
+
+# build with '--no-build-isolation', for better sccache hit rate
+# 0 really means "add --no-build-isolation" (ref: https://github.com/pypa/pip/issues/5735)
+export PIP_NO_BUILD_ISOLATION=0
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+case "${RAPIDS_CUDA_VERSION}" in
+  12.*)
+    EXTRA_CMAKE_ARGS=";-DUSE_CUDA_MATH_WHEELS=ON"
+  ;;
+  11.*)
+    EXTRA_CMAKE_ARGS=";-DUSE_CUDA_MATH_WHEELS=OFF"
+  ;;
+esac
+
+export SKBUILD_CMAKE_ARGS="-DDETECT_CONDA_ENV=OFF${EXTRA_CMAKE_ARGS}"
```

Comment:
Honestly I don't think we need to be setting this explicitly almost ever. All our wheel builds occur in environments where we don't have a conda environment. I'd prefer to remove this altogether.

If we did keep it though, minor nit: I would prefer to have an explicit semicolon here rather than embedding it in each incantation of `EXTRA_CMAKE_ARGS` above.
```suggestion
export SKBUILD_CMAKE_ARGS="${EXTRA_CMAKE_ARGS}"
```

Reply:
Ok that's fine, I can try removing it. Both of those patterns (setting `-DDETECT_CONDA_ENV` and the placement of the semicolon) are used all over RAPIDS wheel-building CI scripts (not newly introduced in this PR), so if that works here and we likes it I'll make other PRs across RAPIDS changing that.

Reply:
Removing it worked for me locally. Did that in https://github.com/rapidsai/raft/pull/2531/commits/1d9e1e97b594c2c84e175f37ab967bb853692255#diff-bf2a37f5233778a838d488772273c05401ae172253b19533bc07e7ca958ba271.

Reply:
Cool thanks. I'll defer to you on when and how you want to propagate these changes through the rest of RAPIDS.

Reply:
> when and how you want to propagate these changes through the rest of RAPIDS

Put up this tracking issue: https://github.com/rapidsai/build-planning/issues/136

I'm going to try to do that in the next day or 2, to try to get it into 25.02.